pallets/file-system/src/utils.rs

```diff
@@ -102,6 +103,15 @@ impl<T> Pallet<T>
 where
     T: pallet::Config,
 {
+    /// Helper function to convert T::AccountId to AccountId32 for signature verification
+    fn account_id_to_account_id32(
+        account_id: &T::AccountId,
+    ) -> Result<sp_core::crypto::AccountId32, Error<T>> {
+        let account_bytes = account_id.encode();
+        sp_core::crypto::AccountId32::try_from(account_bytes.as_slice())
+            .map_err(|_| Error::<T>::InvalidSignature)
+    }
```

Comment:
This would fail in Moonbeam for example since accounts are EVM based and are 20 bytes in length, I believe we should have the signature scheme and account id conversions be configurable types in the pallet config to be agnostic over any chain.